vault/resource_aws_secret_backend_test.go

```diff
@@ -103,6 +103,38 @@ func TestAccAWSSecretBackend_usernameTempl(t *testing.T) {
 	})
 }
 
+func TestAccAWSSecretBackend_remount(t *testing.T) {
+	path := acctest.RandomWithPrefix("tf-test-aws")
+	updatedPath := acctest.RandomWithPrefix("tf-test-aws-updated")
+
+	resourceName := "vault_aws_secret_backend.test"
+	accessKey, secretKey := testutil.GetTestAWSCreds(t)
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testutil.TestAccPreCheck(t) },
+		Steps: []resource.TestStep{
```

Comment:
I was looking in the testing framework to see if there was anything to support checking whether a resource is being destroyed or not between test steps, but I didn't find anything. Do you know if there might be a way to accomplish that? It may be a useful ability to have going forward as we utilize the feature.

Reply:
Good point. I've been meaning to add this check in the Managed keys tests as well, but I think one way would be if Vault has a UUID for the resource then we can check if the UUIDs are different after re-creation. Other than that, I haven't come up with a better solution as of yet unfortunately

Reply:
For this test, I had to check my Vault logs to confirm that a mount migration had occurred successfully

Reply:
There is a `CheckDestroy` field in resource.TestCase that I've seen used elsewhere, though I think that's after the test and not between steps.

You may be able to use `TestCheckNoResourceAttr` to check that the old mount doesn't exist in the state at least?

Reply:
You are correct! `CheckDestroy` runs after the test completes, so unfortunately we can't use that in this case

I looked into trying to make `TestCheckNoResourceAttr` work. The function checks whether certain state keys exist in the TF state for a particular resource. In this case, while the mount does change (expressed in this resource as a `path` change), the `path` key isn't being deleted or removed, so it still exists in the TF state (albeit with a different value). In this case unfortunately, I don't think that would work exactly the way we want it to

I'll definitely keep looking into this as an improvement to our testing since this is also needed in a few other places!

In the future I think we also plan to refactor all secret and auth engines be mounted solely via the `vault_mount` resource (currently some resources mount the engine as well as the config; including the resource we are talking about: https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_aws_secret_backend.go#L119). This might help check the TF state for deleted mounts in a cleaner way.